oauth2_http/java/com/google/auth/oauth2/IdentityPoolSubjectTokenSupplier.java

```diff
@@ -44,8 +44,9 @@ public interface IdentityPoolSubjectTokenSupplier extends Serializable {
   /**
    * Gets a subject token that can be exchanged for a GCP access token.
    *
+   * @param context relevant context from the calling credential.
    * @return a valid subject token.
    * @throws IOException
    */
-  String getSubjectToken() throws IOException;
+  String getSubjectToken(ExternalAccountSupplierContext context) throws IOException;
```

Comment:
The interface is named `IdentityPoolSubjectTokenSupplier`, so it it might be more consistent to name it `IdentityPoolSubjectTokenSupplierContext` or `IdentityPoolSubjectTokenContext`.

Or maybe drop the `IndentityPool` prefix altogether and call it `SubjectTokenSupplier` and `SubjectTokenSupplierContext`?

Reply:
The context object is also used for AWS credential supplier which returns an AWS security credential instead of a subject token so wanted to have something more generic, but can split out into two different context objects - Would look like a parent abstract class called ExternalAccountSupplierContext or possibly just SupplierContext (although I was thinking that might be too generic) and then would have an AwsSecurityCredentialsSupplierContext and SubjectTokenSupplierContext. The only thing is that currently its the exact same information, and thinking about it, I'm not sure if there is any extra information AWS would need that it couldn't get from audience. Not opposed to that really, but just want to make sure there's a reason to split them out

Reply:
I see. I guess the current naming is fine then.